model/historical_test_data.go

```diff
@@ -168,8 +168,8 @@ func (d *HistoricalTestData) Update(ctx context.Context, result TestResult) erro
 		ctx,
 		query,
 		pipeline,
-		options.FindOneAndUpdate().SetReturnDocument(options.After),
```

Comment:
I think we need to keep this in order to update the in-memory stat doc, otherwise `d` will still have the outdated information.

Reply:
my thinking was that it doesn't look like we don't actually use the `d` for anything after the update.
I think I see what you're saying, though: if we start using it we'd expect d to be changed after the update. I'll change it back.

Reply:
done